autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/ui/credential/passwordgeneration/AutofillUseGeneratedPasswordDialogFragment.kt

```diff
@@ -0,0 +1,210 @@
+            if (appBuildConfig.sdkInt <= Build.VERSION_CODES.S_V2) {
+                Toast.makeText(context, R.string.autofillManagementPasswordCopied, Toast.LENGTH_SHORT).show()
+            }
+        }
+
+        with(binding.copyPasswordButton) {
+            post {
+                // actual size is 24dp, but we want to increase touch area to a more comfortable 48dp; adding 12dp to each side
+                val touchableArea = Rect()
+                getHitRect(touchableArea)
+                val extraSpace = 12.toPx()
+                touchableArea.top -= extraSpace
+                touchableArea.bottom += extraSpace
+                touchableArea.left -= extraSpace
+                touchableArea.right += extraSpace
+                (parent as View).touchDelegate = TouchDelegate(touchableArea, this)
+            }
+        }
+    }
+
+    private fun configureGeneratePasswordButton(
+        binding: ContentAutofillGeneratePasswordDialogBinding,
+        originalUrl: String,
+    ) {
+        binding.useSecurePasswordButton.setOnClickListener {
+            pixelNameDialogEvent(GeneratedPasswordAccepted)?.let { pixel.fire(it) }
+
+            val result = Bundle().also {
+                it.putString(UseGeneratedPasswordDialog.KEY_URL, originalUrl)
+                it.putBoolean(UseGeneratedPasswordDialog.KEY_ACCEPTED, true)
+                it.putString(UseGeneratedPasswordDialog.KEY_USERNAME, getUsername())
+                it.putString(UseGeneratedPasswordDialog.KEY_PASSWORD, getGeneratedPassword())
+            }
+            parentFragment?.setFragmentResult(UseGeneratedPasswordDialog.resultKey(getTabId()), result)
+
+            ignoreCancellationEvents = true
+            animateClosed()
+        }
+    }
+
+    private fun configureCloseButton(binding: ContentAutofillGeneratePasswordDialogBinding) {
+        binding.closeButton.setOnClickListener { (dialog as BottomSheetDialog).animateClosed() }
+        binding.cancelButton.setOnClickListener { (dialog as BottomSheetDialog).animateClosed() }
+    }
+
+    override fun onCancel(dialog: DialogInterface) {
+        if (ignoreCancellationEvents) {
+            Timber.v("onCancel: Ignoring cancellation event")
+            return
+        }
+
+        Timber.v("onCancel: GeneratePasswordDialogCancel. User declined to use generated password")
+
+        pixelNameDialogEvent(Dismissed)?.let { pixel.fire(it) }
+
+        val result = Bundle().also {
+            it.putBoolean(UseGeneratedPasswordDialog.KEY_ACCEPTED, false)
+            it.putString(UseGeneratedPasswordDialog.KEY_URL, getOriginalUrl())
+        }
+
+        parentFragment?.setFragmentResult(UseGeneratedPasswordDialog.resultKey(getTabId()), result)
+    }
+
+    private fun animateClosed() {
+        (dialog as BottomSheetDialog).animateClosed()
+    }
+
+    private fun pixelNameDialogEvent(dialogEvent: DialogEvent): AutofillPixelNames? {
```

Comment:
No strong opinion but looks a bit overkill having an interface and a method to fire pixels rather than calling directly `pixel.fire` with the pixel name?

Reply:
Yeah, it is kinda is here. It's a pattern taken from the other autofill dialogs where they have more complicated logic to determine which pixel to send and/or which params to attach. Also have no strong opinions on this case, but will leave it as is to keep it using a consistent approach as with the other autofill dialogs.